operate/common/src/main/java/io/camunda/operate/zeebe/StandalonePartitionSupplier.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.operate.zeebe;
+
+import io.camunda.operate.util.Either;
+import io.camunda.zeebe.client.ZeebeClient;
+
+public class StandalonePartitionSupplier implements PartitionSupplier {
+
+  private final ZeebeClient zeebeClient;
+
+  public StandalonePartitionSupplier(final ZeebeClient zeebeClient) {
+    this.zeebeClient = zeebeClient;
+  }
+
+  @Override
+  public Either<Exception, Integer> getPartitionsCount() {
+    try {
+      final var topology = zeebeClient.newTopologyRequest().send().join();
+      final var partitionCount = topology.getPartitionsCount();
+      return Either.right(partitionCount);
+    } catch (Exception t) {
+      return Either.left(t);
```

Comment:
who said that left is bad and right is good ? 😄

Reply:
I think this is a pattern we use across the stack; at least, I am aware of how Zeebe uses such utilities.